### ISSUE TITLE
Compilation error when no GUI library is available

### ISSUE THREAD

**Issue by rolalaro** [lagadic/visp#1404]
[ViSP-third-party.txt](https://github.com/lagadic/visp/files/15337327/ViSP-third-party.txt)

Dear contributors,

I faced the following compilation error when I tried to compile ViSP after deactivating the GUI libraries:
```
In file included from /home/user/visp_ws/visp/modules/core/include/visp3/core/vpImageFilter.h:51,
                 from /home/user/visp_ws/visp/modules/core/include/visp3/core/vpCannyEdgeDetection.h:41,
                 from /home/user/visp_ws/visp/modules/core/src/image/vpCannyEdgeDetection.cpp:33:
/home/user/visp_ws/visp/modules/core/include/visp3/core/vpImageConvert.h:151:32: error: ‘std::mutex’ has not been declared
  151 |                                std::mutex *pointcloud_mutex = nullptr,
      |                                ^~~
/home/user/visp_ws/visp/modules/core/include/visp3/core/vpImageConvert.h:156:32: error: ‘std::mutex’ has not been declared
  156 |                                std::mutex *pointcloud_mutex = nullptr,

```

Here are the steps I followed that led to this error:
```
$ cd ~/visp_ws/visp-build
$ rm -rf *
$ ccmake ../visp -DUSE_X11=OFF -DUSE_OPENCV=OFF
$ make -j10
```

I think that vpImageConvert is missing the following line:
```
#include <mutex>
```

Regards

**Comment by rolalaro** [comment]
Proposition of a fix in PR #1405

**Comment by rolalaro** [comment]
The problem has been addressed in PR #1405 . Some improvements suggested by @SamFlt  can be done in the vpDisplayFactory .
